src/queue.c

```diff
@@ -861,6 +861,14 @@ gettid(void)
 		if ((f) && tsd->k) ((void(*)(void*))(f))(tsd->k); \
 	} while (0)
 
+#ifdef __ANDROID__
+static void (*_dispatch_thread_detach_callback)(void);
```

Comment:
if it's static then the overlay which is a dynamic library can't pick it up.

@das suggested to me the right place is to export the symbol is from `private/queue_private.h` or `private/private.h` and also to use silgen like you did.